src/controllers/controllerengine.cpp

```diff
@@ -842,12 +841,6 @@ QScriptValue ControllerEngine::connectControl(
 
     if (function.isFunction()) {
         qDebug() << "Connection:" << group << name;
-        connect(cos, SIGNAL(valueChanged(double)),
-                this, SLOT(slotValueChanged(double)),
-                Qt::QueuedConnection);
-        connect(cos, SIGNAL(valueChangedByThis(double)),
```

Comment:
ControlObjectScript::slotValueChanged() ignores the source object. So the script function is called even if the value was changed by the script.